.github/workflows/nightly_detect_failure.yml

```
@@ -41,7 +41,15 @@ jobs:
         if [[ $toolchain =~ leanprover/lean4:nightly-([a-zA-Z0-9_-]+) ]]; then
           version=${BASH_REMATCH[1]}
           echo "NIGHTLY=$version" >> $GITHUB_ENV
-          git push origin refs/heads/nightly-testing:refs/heads/nightly-testing-$version
+          # Check if the remote tag exists
+          if git ls-remote --tags origin refs/tags/nightly-testing-$version | grep -q 'refs/tags/nightly-testing-$version'; then
```

Minor suggestion, avoiding grep, and `--tags` means one does not have to write `refs/tags`, I believe
```suggestion
          if git ls-remote --tags --exit-code origin "nightly-testing-$version" >/dev/null; then
```